File: 6.1D/main/java/com/example/personalizedlearningexperiences/UserDatabase.java
package com.example.personalizedlearningexperiences;

import androidx.room.Database;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

@Database(entities = {User.class}, version = 2)
@TypeConverters({Converters.class, QuestionConverter.class})
public abstract class UserDatabase extends RoomDatabase {
    public abstract UserDAO userDao();
}

File: 5.1C iTube/java/com/example/itube/UserDatabase.kt
package com.example.itube

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import com.example.itube.UserDAO

@Database(entities = [User::class], version = 1)
@TypeConverters(Converters::class) // Add this line to include the TypeConverter
abstract class UserDatabase: RoomDatabase() {

    abstract fun userDao(): UserDAO

}